### PR TITLE
ifconfig is deprecated

### PR DESCRIPTION
**Pull request by sardaukar**
As seen here https://ubuntu.com/blog/if-youre-still-using-ifconfig-youre-living-in-the-past (from 2017)

### CODE REVIEW

**Comment by woudsma** [comment]
@sardaukar Interesting. Yeah so using `ip` would be better.

I've just tried on my DigitalOcean VPS:
```sh
ip -o route show to default | awk '{print $3}'
```
Unfortunately it prints the wrong IP address. I guess this doesn't work for both DO and Vultr.

**Comment by sardaukar** [comment]
what's the output of `ip -o route show to default` ?

**Comment by woudsma** [comment]
```sh
default via 178.62.192.1 dev eth0 proto static
```
The actual IP address is 178.62.203.115.
I cannot find 178.62.192.1 anywhere in the output of `ifconfig`.

**Comment by sardaukar** [comment]
That's weird. Are you sure the actual address is 178.62.203.115? (also, are you on IRC or something? :D )

**Comment by woudsma** [comment]
@sardaukar Pretty sure 😒

**Comment by woudsma** [comment]
It's quite weird, I don't see 178.62.192.1 mentioned anywhere on the server, not in the `ifconfig` output and not in the `ip address` output

**Comment by sardaukar** [comment]
what about ` awk '/32 host/ { print f } {f=$2}' <<< "$(</proc/net/fib_trie)"` ?

**Comment by woudsma** [comment]
Yeah, that prints a bunch of IP addresses, using this I have the server IP:
```sh
awk '/32 host/ { print f } {f=$2}' <<< "$(</proc/net/fib_trie)" | tail -n 1
```
Does it have to be such a complicated expression? 😛

**Comment by sardaukar** [comment]
It's weird the IP is nowhere else...

File: src/subcommands/init [hunk 1]
```diff
@@ -25,7 +25,7 @@ cmd-init() {
   echo "$PREFIX Selecting network interface"
   ETH_LINK=$(get-network-interface-name)
 
-  NODE_IP=$(ifconfig $ETH_LINK | grep -oP 'inet \K\S+')
+  NODE_IP=$(ip -o route show to default | awk '{print $3}')
```
Comment (woudsma) [inline src/subcommands/init:28]: We need to find a cross-platform solution for getting the IP address of the host.
This command returns a wrong IP address when using DigitalOcean.

File: src/subcommands/init [hunk 2]
```diff
@@ -25,7 +25,7 @@ cmd-init() {
   echo "$PREFIX Selecting network interface"
   ETH_LINK=$(get-network-interface-name)
 
-  NODE_IP=$(ip -o route show to default | awk '{print $3}')
+  NODE_IP=$(curl -L ipconfig.me)
```
Comment (woudsma) [inline src/subcommands/init:28]: Thanks for the suggestion, that works.
It's not the most stable solution, to be dependant on the response of an external service for initializing the swarm - which is quite critical.

What is the output of this on Vultr?
```sh
hostname -I | awk '{print $1}'
```

Comment (sardaukar) [reply src/subcommands/init:28]: Yes, it does.

Comment (woudsma) [reply src/subcommands/init:28]: Let's use that for the moment, agree?

**Pull request merged**